testutil/test.go

```diff
@@ -114,9 +166,56 @@ func RandFact(domain string, tx uint64) *origins.Fact {
 	}
 }
 
+// RandDictionaryFact returns a metod for generating facts from a given dictionary of values
```

Comment:
"RandDictionaryFact returns a `GeneratorFunc`...". Also the return value should be `GeneratorFunc` rather than the full type signature.